### ISSUE TITLE
Duplicate broker logins are incorrectly allowed

### ISSUE THREAD

**Issue by jecollins** [powertac/powertac-server#804]
The setup process for a sim session involves specifying a list of authorized broker names and then waiting for all of them to login (with a possible timeout) before starting the session. However, once the list of authorized broker usernames is constructed, it is used for validating the names of incoming login requests, but duplicate names are not detected, and the session starts when the number of validated logins matches the length of the list of authorized usernames. In other words, the names are not "checked off" the list as they log in. The result is that if there are two brokers with the same valid username, the server will count them as valid logins and will start before the last configured broker has logged in.

**Comment by jecollins** [comment]
We have confirmed that this cannot happen in a tournament situation, because brokers need registration credentials from the tournament manager before then are allowed to log in to the server.

**Comment by jecollins** [comment]
Low priority.

**Comment by jecollins** [comment]
This has been fixed, apparently some time ago. See the last few lines of CompetitionControlService.loginBroker().
